crates/bevy_core_pipeline/src/contrast_adaptive_sharpening/mod.rs

```diff
@@ -0,0 +1,279 @@
+use crate::{core_2d, core_3d, fullscreen_vertex_shader::fullscreen_shader_vertex_state};
+use bevy_app::prelude::*;
+use bevy_asset::{load_internal_asset, HandleUntyped};
+use bevy_ecs::{prelude::*, query::QueryItem};
+use bevy_reflect::{Reflect, TypeUuid};
+use bevy_render::{
+    extract_component::{ExtractComponent, ExtractComponentPlugin, UniformComponentPlugin},
+    prelude::Camera,
+    render_graph::RenderGraph,
+    render_resource::*,
+    renderer::RenderDevice,
+    texture::BevyDefault,
+    view::{ExtractedView, ViewTarget},
+    Render, RenderApp, RenderSet,
+};
+
+mod node;
+
+pub use node::CASNode;
+
+const SHARPEN_MIN: f32 = 0.0;
+// I didn't really notice a difference after 2.0, but it might just be the scene I tested it on.
+const SHARPEN_MAX: f32 = 4.0;
+
+/// Applies a contrast adaptive sharpening (CAS) filter to the camera.
+///
+/// CAS is usually used in combination with shader based anti-aliasing methods
+/// such as FXAA or TAA to regain some of the lost detail from the blurring that they introduce.
+///
+/// CAS is designed to adjust the amount of sharpening applied to different areas of an image
+/// based on the local contrast. This can help avoid over-sharpening areas with high contrast
+/// and under-sharpening areas with low contrast.
+///
+/// To use this, add the [`ContrastAdaptiveSharpeningSettings`] component to a 2D or 3D camera.
+#[derive(Component, Reflect, Clone)]
+#[reflect(Component)]
+pub struct ContrastAdaptiveSharpeningSettings {
+    /// Enable or disable sharpening.
+    pub enabled: bool,
+    /// Adjusts sharpening strength. Higher values reduce the amount of sharpening.
+    ///
+    /// 0.0 is full strength, 1.0 is half strength, 2.0 is quarter strength, etc.
+    /// Clamped between 0.0 and 4.0.
+    ///
+    /// The default value is 0.2.
+    pub sharpening_strength: f32,
+    /// Whether to try and avoid sharpening areas that are already noisy.
+    ///
+    /// You probably shouldn't use this, and just leave it set to false.
+    /// You should generally apply any sort of film grain or similar effects after CAS
+    /// and upscaling to avoid artifacts.
```

Comment:
Perhaps it is useful if you have noisy textures?

Reply:
The docs seemed to imply that it's mainly for when you're integrating it into an existing engine and you might not be able to insert it into the pipeline before eg. film grain is applied.

Reply:
Ok. :)